unit_tests/test_ssh_server.py

```diff
@@ -0,0 +1,76 @@
+import os
+import pytest
+import subprocess
+import time
+import socket
+
+from sdcm.remote.remote_cmd_runner import RemoteCmdRunnerBase
+from sdcm.utils.decorators import retrying
+
+
+@pytest.fixture(scope="module")
+def ssh_server_container(tmp_path_factory):
+    exposed_port = 22
+    dockerfile_content = f"""
+    FROM ubuntu:25.10
+    RUN apt-get update && apt-get install -y openssh-server
+    RUN echo 'root:root' | chpasswd
+    RUN useradd -m test
+    RUN passwd -d test
+    RUN echo 'PermitRootLogin yes' > /etc/ssh/sshd_config \
+        && echo 'PasswordAuthentication yes' >> /etc/ssh/sshd_config \
+        && echo 'PermitEmptyPasswords yes' >> /etc/ssh/sshd_config \
+        && echo 'AuthenticationMethods none' >> /etc/ssh/sshd_config \
+        && echo 'UsePAM no' >> /etc/ssh/sshd_config
+    EXPOSE {exposed_port}
+    CMD ["/usr/sbin/sshd", "-D"]
+    """
+
+    tmpdir = tmp_path_factory.mktemp("ssh_server")
+    dockerfile_path = os.path.join(tmpdir, "Dockerfile")
+    with open(dockerfile_path, "w") as f:
+        f.write(dockerfile_content)
```

Comment:
The fact that you need to write static info to a file means this file should be existing in the file tree as part of the project.